### DISCUSSION TITLE
Override default figure numbering?

### DISCUSSION THREAD

**Discussion by obrl-soil** [quarto-dev/quarto-cli#1806] [Q&A] [answered]
Is there a way to override default figure and table numbering? I'm Quarto-ising a book with no native chapter numbering, so the figures and tables should just be numbered 1:n over the whole document in order to match the hard-copy.

I'm using a combination of linked images (not plots) and flextables generated from code chunks.

**Comment by mcanouil** [comment]
Not clear, what you mean by override default figure and table numbering.

It does not matter that the tables/figures are generated or simply included, if they have captions they will get numbered.

Did you look at <https://github.com/quarto-dev/quarto-cli/discussions?discussions_q=numbering> and <https://github.com/quarto-dev/quarto-cli/discussions?discussions_q=numbering>?

**Comment by rgaiacs** [comment]
The problem that @obrl-soil described is cover by the following minimal working example:

`_quarto.yml`:

```
project:
  type: book

book:
  title: "Minimal Working Example"
  chapters:
    - index.qmd
    - ch1.qmd
    - ch2.qmd

format:
  html:
    theme: cosmo
``` 

`ch1.qmd`:

``` 
# Foo {.unnumbered}

Some text

![First image](cover.png){#fig-foo}
```

`ch2.qmd`:

``` 
# Bar {.unnumbered}

Some text

![Second image](cover.png){#fig-bar}
```

The expected HTML output is that figures will numbered 1 and 2 but the observed output is 1 and 1.

![Screenshot from 2022-08-11 16-53-36](https://user-images.githubusercontent.com/1506457/184098790-5754bc53-2132-47b2-8097-24c50b9c0bbd.png)

My understanding is that Quarto doesn't keep track of the number of figures in each file and because of it can't provide the starting number to Pandoc.

**Reply by lauren-obrien** [reply]
Yes, thanks.

**Comment by mcanouil** [comment]
Ok, so the actual question is "how to get continuous figures/tables numbering across chapters?"
I don't think it's currently possible without post-processing with a Lua filter.

**Comment by jjallaire** [comment, marked as answer]
I don't think even post-processing would work here as we are doing crossref resolution in our own post-processor. So currently there is no way to do this however we will be doing additional work on making crossrefs more customizable soon and we can address this requirement then.

**Comment by maia-sh** [comment]
Has continuous numbers across (book) sections been resolved in later versions, and if so how?

**Reply by mcanouil** [reply]
The issue attached to this is still open (you can see it in the right sidebar of this discussion).

- https://github.com/quarto-dev/quarto-cli/issues/1818
